agent/lib/kontena/rpc_client.rb

```diff
@@ -6,13 +6,16 @@ class RpcClient
     include Kontena::Logging
 
     class Error < StandardError
-      attr_reader :code
+      attr_reader :code, :remote_backtrace
 
       def initialize(code, message, remote_backtrace = nil)
         @code = code
+        @remote_backtrace = (['<Remote>:'] + remote_backtrace) if remote_backtrace
         super(message)
-        set_backtrace(caller) unless backtrace
-        set_backtrace(backtrace + (['<Remote backtrace>'] + Array(remote_backtrace))) if remote_backtrace
+      end
+
+      def backtrace
+        (super || caller) + Array(remote_backtrace)
```

Comment:
Does `super` aka `Exception#backtrace` ever return nil? Using the `caller` of the `#backtrace` method does not sound like the right thing to return O_o